_posts/2020-03-05-snowplow-android-tracker-1.4.0-released/index.md

```diff
@@ -0,0 +1,169 @@
+|--------------:|:----------------------------|:--------------|:---------|
+|`basisForProcessing` | GDPR Basis for processing | Yes           | Enum String   |
+|     `documentId` | ID of a GDPR basis document     | No           | String   |
+|        `documentVersion` | Version of the document        | No            | String   |
+| `documentDescription` | Description of the document | No            | String   |
+
+The required basisForProcessing accepts only the following literals: `consent`, `contract`, `legal_obligation`, `vital_interests`, `public_task`, `legitimate_interests` - in accordance with the [five legal bases for processing](https://ico.org.uk/for-organisations/guide-to-data-protection/guide-to-the-general-data-protection-regulation-gdpr/lawful-basis-for-processing/)
+
+The GDPR context is enabled by calling the `gdprContext` method of the tracker builder or by calling the `enableGdprContext` method once the tracker has been initialised.
+
+It is called as follows:
+
+{% highlight java %}
+Tracker.instance().enableGdprContext(
+    Gdpr.Basis.CONSENT,
+    "someId",
+    "0.1.0",
+    "a demo document description"
+);
+{% endhighlight %}
+
+
+<h2 id="diagnostic">2. Diagnostic feature for internal error tracking</h2>
+
+The tracker is constantly tested with different platforms and versions but there are rare situations where an error can be reported. The tracker manages errors internally, avoiding crashes of the app and assuring that the events are not lost. However, it can now report errors to a Snowplow collector as `diagnostic_error` events.
+
+To activate this feature you only need to enable `trackerDiagnostic` ([#343][343]):
+
+{% highlight java %}
+TrackerBuilder trackerBuilder =
+    new TrackerBuilder(emitter, namespace, appId, appContext)
+        .trackerDiagnostic(true)
+        ...
+        .build();
+Tracker.init(trackerBuilder);
+{% endhighlight %}
+
+It will automatically report any internal tracker error to the configured Snowplow collector. These will be fundamental for the troubleshooting unexpected behaviour of the tracker.
+For Snowplow Insights customers, the sequence of tracker diagnostic events can be sent to the Snowplow support team for analysis in order to spot tracker issues or the source of unexpected behaviour.
+
+“unstruct_event_com_snowplowanalytics_snowplow_diagnostic_error_1”: {
+      “className”: “Tracker”,
+      “message”: “pool-1-thread-2|Error tracker message”
+}
+
+
+<h2 id="iab">3. Fix for IAB enrichment identification</h2>
+
+As reported [here](https://discourse.snowplowanalytics.com/t/warning-iab-enrichment-treats-android-tracker-events-as-spider-generated/2482) the IAB enrichment incorrectly treats events sent by the Android tracker as spider-generated. This is due to the fact that the default User Agent used by the Android tracker, which is the one of the underlying library used to make HTTP calls (okHttp), is part of the IAB blacklist.
```

Comment:
This sentence is a bit confusing: 
This is due to the fact that the default User Agent used by the Android tracker, which is the one of the underlying library used to make HTTP calls (okHttp), is part of the IAB blacklist.

Does it mean: 
This is due to the fact that the underlying library used to make HTTP calls (okHttp) for the default User Agent used by the Android tracker is part of the IAB blacklist.

Reply:
I think we can omit all these details. The real meaning is:
The Android Tracker uses okHttp library. The default User-Agent of that library is in the IAB blacklist. Hence the Android tracker is seen as spider-generator.

What if I just say _This is due to the fact that the default User Agent used by the Android tracker is part of the IAB blacklist due to the underlying library `okHttp`._ ?

Reply:
My 2c - This is actually the clearest explanation (just changing to past tense to denote it's fixed now):

`The Android Tracker uses okHttp library. The default User-Agent of that library is in the IAB blacklist. Hence the Android tracker was seen as spider-generator.`

Reply:
Or: 

`The Android Tracker uses the okHttp library. The default User-Agent of that library is in the IAB blacklist, which caused the Android tracker to be seen as spider-generator.`